src/components/Button/styles.scss

```diff
@@ -99,6 +99,10 @@
     &[disabled] {
         background-color: $color-gray-1;
         border-color: $color-gray-1;
+
+        &:hover {
+            color: $color-gray-2;
+        }
```

Comment:
you don't need the pseudo-clase :hover
you only have add the ```color: $color-gray-2;``` to ```&[disabled]```
anything like that
```
&[disabled] {
        background-color: $color-gray-1;
        border-color: $color-gray-1;
        color: $color-gray-2;
}
```